### ISSUE TITLE
Collectors can add description text on resources they collect

### ISSUE THREAD

**Issue by toobulkeh** [vaporware/parkland#11]
Ala Steam (www.steamcommunity.com) Curators.

**Comment by toobulkeh** [comment]
@hsidar this might be of interest to you for starters

**Comment by hsidar** [comment]
![screenshot_081215_021452_pm](https://cloud.githubusercontent.com/assets/11035160/9232590/82380936-40fc-11e5-8b86-2fe402a5c4a4.jpg)

Is that what you are looking for?

**Comment by toobulkeh** [comment]
The design will probably change, but yeah, essentially a paragraph there. Here's what I imagine it looks like at scale:

http://store.steampowered.com/curator/1370293-Cynical-Brit-Gaming/

**Comment by hsidar** [comment]
Gotcha. So, thinking through this, my initial thoughts are that you already have the has_many through: association and resource_collection model set up. So questions:

- Would the description go into that model? 
- If so, then past that would it need to be in the include statement in the collections controller, or is what is there enough?
`@resource_collections = @collection.resource_collections.includes(:resource).order("position")`

Past that, the times the description would be interacted with:

- When user collects resource (prompted to create description).
- When viewing collection.
- When editing collection/resource.

Seem about right?

**Comment by toobulkeh** [comment]
Sounds like you got it! Right now there is no functionality to edit a collection, so that page should be interesting!

**Comment by hsidar** [comment]
Pull request out. Judge me.

I saw that the action to add resources to collections was just a checkbox basically. So, in order to prompt for description there, I would have to add a popup or that would have to change entirely.

So instead of doing anything disruptive like that, I just set the description on the resource collections to default to the description on the resource that was already there. I figure in most cases people would be satisfied with the default description and would only customize it when they wanted to add something. This way, users can rip and run and edit things later without worrying about a blank entry, or a prompt every time they add resources to a collection.

When a user goes to a collection, the field for resource descriptions is editable there if they own the collection. I didn't put any fancy notice when it updates, because you said that was all gonna change. There is a console notice of success though. I tried to do a quick snap to green, then fade back to white using jquery, but apparently jquery-ui doesnt come packaged with jquery colors? Dunno, couldn't get it to work if it was there.

Let me know if I missed anything or did anything naive. I learned a lot with this and hopefully it was helpful.
